### ISSUE TITLE
Publish MarinDNA m5.1 on Hugging Face and create the blog Collection

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/marin-dna#389]
🤖 Publish the headline MarinDNA m5.1 base model, with its tokenizer bundled, to the [`bolinas-dna`](https://huggingface.co/bolinas-dna) Hugging Face organization. Create a public, blog-specific Hugging Face Collection containing the model, the five training datasets, their five matched validation datasets, and the Mendelian and SGE evaluation datasets.

This is a native sub-issue of #361. It deliberately keeps the first release small and polished. Other models, checkpoints, probes, dashboards, and evaluation-result artifacts are later work and do not block this issue.

## Goal

Release:

- **Model:** m5.1 1B base genomic language model
- **Source run:** `dna-bolinas-mix-v0.9-p1B-i24-exp135-zoonomia-m5.1-bef41e`
- **Source checkpoint:** `hf/step-59158`
- **Repository:** `bolinas-dna/marin-dna-exp135-m5.1`
- **Tokenizer:** bundled in the model repository
- **Access:** public and ungated
- **License:** Apache-2.0

The model page should become the canonical m5.1 link used by the blog and its diagrams.

## Publish the m5.1 base model

- [x] Confirm the source checkpoint and complete file inventory.
- [x] Draft the model card for human review before any Hugging Face upload.
- [x] Create the public `bolinas-dna/marin-dna-exp135-m5.1` repository.
- [x] Upload the final m5.1 base-model checkpoint to `main`.
- [x] Bundle the complete tokenizer and tokenizer configuration.
- [x] Apply the Apache-2.0 license metadata and license file.
- [x] Record source and destination identifiers, byte sizes, and checksums.
- [x] Verify standard one-call loading:

```python
from transformers import AutoModelForCausalLM, AutoTokenizer

repo_id = "bolinas-dna/marin-dna-exp135-m5.1"

tokenizer = AutoTokenizer.from_pretrained(repo_id)
model = AutoModelForCausalLM.from_pretrained(repo_id)
```

## Create the blog Collection

Create a public Collection in the `bolinas-dna` organization titled:

**MarinDNA - Genomic Language Model Optimization Blog 2026 07**

Add exactly the following resources.

### Model

- m5.1 1B base model

### Five training datasets

- [CDS](https://huggingface.co/datasets/bolinas-dna/genomes-v5-genome_set-animals-intervals-v5_255_128)
- [Upstream](https://huggingface.co/datasets/bolinas-dna/genomes-v5-genome_set-animals-intervals-v1_255_128)
- [Downstream](https://huggingface.co/datasets/bolinas-dna/genomes-v5-genome_set-animals-intervals-v15_255_128)
- [Enhancer](https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-v3_ccre_non_promoter)
- [ncRNA](https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-v3_ncrna_exon)

### Five matched validation datasets

- [CDS validation](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v5_255_255)
- [Upstream/promoter validation](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v1_255_255)
- [Downstream validation](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v15_255_255)
- [Enhancer validation](https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-val_enhancer)
- [ncRNA validation](https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-val_ncrna)

### Evaluation datasets

- [Mendelian](https://huggingface.co/datasets/bolinas-dna/evals_mendelian_traits)
- [SGE](https://huggingface.co/datasets/bolinas-dna/evals_sge)

The Collection description should link to the blog and identify:

- m5.1 as the model released with the blog;
- the five training resources as training datasets;
- the five matched datasets as training-validation probes; and
- Mendelian and SGE as downstream evaluation-only datasets.

The Collection must not imply that all five training datasets contributed to every m5.1 training phase, that the matched validation probes were training data, or that Mendelian and SGE scores are level-comparable.

Do not add dashboards, leaderboard pages, additional evaluation datasets, or evaluation-result artifacts.

## Model card

The reviewed model card must include:

- a 1–2 sentence description and provenance;
- architecture, parameter count, context length, checkpoint step, and approximate token exposure;
- tokenizer, BOS/EOS convention, and input format;
- the sequence of training mixtures in the m5.1 lineage;
- links to the five training datasets and five matched validation datasets;
- intended uses, limitations, and evaluation caveats;
- the Apache-2.0 license;
- a commit-pinned link to the [training script](https://github.com/marin-community/marin/blob/a41a83fdddf85a75e39b56c32949518e3f578/experiments/dna/exp135_bolinas_mix_sweep.py#L514-L657);
- links to the blog, #361, #370, Mendelian, and SGE;
- the minimal loading example;
- a statement that this is the final m5.1 checkpoint at step 59158; and
- the tags `biology`, `genomics`, and `dna`.

Do not claim that m5.1 significantly beats Evo 2 40B or AlphaGenome; #370 records the supported interpretation.

## Verification

Create a tracked manifest and verification script under `scripts/issue<N>/` on the issue's permanent, unmerged branch.

- [x] Record the exact source GCS URI, file inventory, sizes, and checksums.
- [x] Record model configuration, parameter count, tokenizer vocabulary, and special tokens.
- [x] Record the exact Hugging Face IDs and current revisions of all Collection datasets.
- [x] Fail on missing files, checksum mismatches, unexpected parameter counts, or tokenizer mismatches.
- [x] Confirm the model and tokenizer download without credentials.
- [x] Load both from the public repository in a clean environment.
- [x] Run a deterministic inference smoke test and compare with the source checkpoint within the expected tolerance.
- [x] Re-fetch the model card and Collection and verify rendering, links, tags, grouping, and item ordering.
- [x] Record the final Hugging Face commit hash in the issue.

## Blog hand-off

- [x] Replace the placeholder m5.1 link in #368.
- [x] Add the Collection link where the blog refers to these resources as a group.
- [x] Update the blog workspace tracked by #373 ([draft PR #390](https://github.com/Open-Athena/marin-dna/pull/390)).
- [ ] Use #367 for the final website transfer.

## Explicitly later — non-blocking

- m5.1 intermediate checkpoints;
- m1.3 and m3.3;
- the parameter-scaling ladder and its intermediate checkpoints;
- the Figure 9 mixture sweep;
- other historical MarinDNA checkpoints; and
- multi-model release automation.

The final m5.1 base model, bundled tokenizer, blog Collection, and verified links are sufficient to close this issue.

## Non-goals

- Publishing any checkpoint other than final m5.1.
- Publishing fitted probes, probe-training artifacts, dashboards, or evaluation-result artifacts.
- Adding downstream evaluation datasets other than Mendelian and SGE.
- Recreating missing checkpoints.
- Changing model weights, evaluations, figures, or blog conclusions.
- Merging the release script or manifest into `main`.
- Cleaning up unrelated Hugging Face repositories.

## Acceptance criteria

- [x] Final m5.1 is public, ungated, Apache-2.0 licensed, and directly loadable from `bolinas-dna/marin-dna-exp135-m5.1`.
- [x] Its tokenizer is bundled and loadable from the same repository.
- [x] The model card is reviewed, provenance-complete, and correctly tagged.
- [x] Source and destination inventories and checksums are verified.
- [x] Clean-environment loading and deterministic inference pass.
- [x] The public Collection is titled **MarinDNA - Genomic Language Model Optimization Blog 2026 07**.
- [x] The Collection contains m5.1, the five training datasets, their five matched validation datasets, and only Mendelian and SGE as downstream evaluation datasets.
- [x] The Collection clearly distinguishes training, validation, and downstream evaluation resources.
- [x] The Collection contains no dashboards or evaluation-result artifacts.
- [ ] Blog and diagram links resolve to the verified model or Collection.
- [x] Other checkpoints remain explicitly non-blocking follow-up work.

**Comment by gonzalobenegas** [comment]
🤖 The review-gated m5.1 release draft is ready on the permanent, intentionally unmerged branch `claude/issue-389-publish-m51` at commit `220e7120ca22a05025fee8a9129927357f78a4d2`.

**Human review required before any Hugging Face write:** [model-card draft](https://github.com/Open-Athena/marin-dna/blob/220e7120ca22a05025fee8a9129927357f78a4d2/scripts/issue389/model_card.md#L1-L114)

Supporting artifacts:

- [Source/release manifest](https://github.com/Open-Athena/marin-dna/blob/220e7120ca22a05025fee8a9129927357f78a4d2/scripts/issue389/manifest.json#L1-L434)
- [Fail-fast verification script](https://github.com/Open-Athena/marin-dna/blob/220e7120ca22a05025fee8a9129927357f78a4d2/scripts/issue389/verify_release.py#L1-L419)
- [Collection description](https://github.com/Open-Athena/marin-dna/blob/220e7120ca22a05025fee8a9129927357f78a4d2/scripts/issue389/collection_description.md#L1-L8)

Verified before this review gate:

- canonical GCS and evals_v2 S3 inventories agree on the five checkpoint files (4,483,118,239 bytes); all five are MD5-identical and now have provider-independent SHA-256 values;
- exact source structure is 1,120,772,224 float32 parameters across 212 tensors, with the expected 7-token tokenizer and BOS/no-EOS convention;
- deterministic CPU inference matches the recorded source logits;
- all twelve Collection datasets are public, ungated, and pinned at their current revisions;
- the requested model repo and Collection title are currently unused.

Three provenance/hand-off notes:

- The training-script SHA in the issue omitted `de`; the draft uses the verified public commit `a41a83fdddfdef85a75e39b56c32949518e3f578`.
- Standard loading succeeds, but Transformers emits a source-config RoPE validation warning (`original_max_position_embeddings=8192` with a 256-token context). The checkpoint/config remain byte-for-byte unchanged to preserve source logits.
- The canonical blog URL is not live yet (currently 404); #367 remains the final website hand-off.

No Hugging Face model repository or Collection has been created. Please approve the model card or request edits here.

**Comment by gonzalobenegas** [comment]
🤖 The approved MarinDNA m5.1 release is now public:

- [Model: `bolinas-dna/marin-dna-exp135-m5.1`](https://huggingface.co/bolinas-dna/marin-dna-exp135-m5.1)
- [Blog Collection: MarinDNA - Genomic Language Model Optimization Blog 2026 07](https://huggingface.co/collections/bolinas-dna/marindna-genomic-language-model-optimization-blog-2026-07-6a627428aa94a4960c855072)
- Final Hugging Face model revision: `c0676b2012b8b9c526deb26ff517f6b92b6d375d`

The permanent release record is commit `710966b28f3d2eb50470926d66c495ff823fb018`:

- [Final manifest and revisions](https://github.com/Open-Athena/marin-dna/blob/710966b28f3d2eb50470926d66c495ff823fb018/scripts/issue389/manifest.json#L1-L449)
- [Fail-fast public verifier](https://github.com/Open-Athena/marin-dna/blob/710966b28f3d2eb50470926d66c495ff823fb018/scripts/issue389/verify_release.py#L1-L424)
- [Approved model card](https://github.com/Open-Athena/marin-dna/blob/710966b28f3d2eb50470926d66c495ff823fb018/scripts/issue389/model_card.md#L1-L114)

Verification passed:

- public, ungated, Apache-2.0 model page with the required `biology`, `genomics`, and `dna` tags;
- exact source file sizes/hashes, including server-side SHA-256 for the 4.18 GiB safetensors file;
- anonymous snapshot download with `token=False`;
- direct `AutoTokenizer` and `AutoModelForCausalLM` loading in an isolated environment with implicit Hugging Face tokens disabled;
- 1,120,772,224 parameters and deterministic source-vs-public logits with maximum absolute difference `0.0`;
- all twelve dataset revisions remain public and pinned;
- exactly 13 Collection items in the reviewed order, with training, validation-probe, and downstream-evaluation notes rendered;
- both public pages return HTTP 200.

The canonical blog URL is still pending the #367 website hand-off, so I am leaving this issue open while I update the #368/#373 link surfaces. I will not close it without explicit approval.

**Comment by gonzalobenegas** [comment]
🤖 Blog hand-off is staged and routed:

- #368 now links m5.1 directly to the public model, links the blog Collection, and keeps the scaling-ladder / other multi-model releases explicitly pending.
- [Draft PR #390](https://github.com/Open-Athena/marin-dna/pull/390) targets claude/issue-373-blog-staging, as required by the blog workspace.
- [Article update](https://github.com/Open-Athena/marin-dna/blob/d2447a7253e28ce0662886ecf16e467e4cb81825/blog/genomic-lm-optimization/content/blog/genomic-lm-optimization.md#L55) and [m5.1 SVG update](https://github.com/Open-Athena/marin-dna/blob/d2447a7253e28ce0662886ecf16e467e4cb81825/blog/genomic-lm-optimization/static/assets/images/blog/genomic-lm-optimization/mini_m5.1_lineage.svg#L1-L34) pass the workspace validator and pinned real-site build.
- The final transfer packet is recorded on [#367](https://github.com/Open-Athena/marin-dna/issues/367#issuecomment-5063213185).

The issue checklist now leaves only the final #367 transfer and post-transfer live-link verification open. I am leaving #389 open and have not closed or merged anything.

**Comment by gonzalobenegas** [comment]
🤖 Closing as completed at the request of @gonzalobenegas. The [m5.1 model](https://huggingface.co/bolinas-dna/marin-dna-exp135-m5.1) and [13-item blog Collection](https://huggingface.co/collections/bolinas-dna/marindna-genomic-language-model-optimization-blog-2026-07-6a627428aa94a4960c855072) are public and verified, with the permanent release record at [commit 710966b](https://github.com/Open-Athena/marin-dna/tree/710966b28f3d2eb50470926d66c495ff823fb018/scripts/issue389). Independent review found no release correctness defect. Final blog linking is explicitly handed back to Gonzalo and is non-blocking for this release.
